### ISSUE TITLE
Error when using http requests queue

### ISSUE THREAD

**Issue by piernik** [Foxandxss/angular-toastr#44]
Hi

I'm using You script -it's great.
I've got one problem.
If I add that code to my app config I always get error on first toaster
```
TypeError: undefined is not a function
    at http://root.system.local/bower_components/angular-toastr/dist/angular-toastr.tpls.js:142:30
    at processQueue (http://root.system.local/bower_components/angular/angular.js:13170:27)
```
Here is code - it is responsible for queueing requests to server.

```
.config(['$httpProvider', function ($httpProvider) {
            /**
             * Interceptor to queue HTTP requests.
             */
            $httpProvider.interceptors.push(['$q', function ($q) {
                var _queue = [];

                /**
                 * Executes the top function on the queue (if any).
                 */
                function _executeTop() {
                    if (_queue.length === 0) {
                        return;
                    }
                    _queue[0]();
                }

                return {
                    /**
                     * Blocks each request on the queue. If the first request, processes immediately.
                     */
                    request: function (config) {
                        var deferred = $q.defer();
                        _queue.push(function () {
                            deferred.resolve(config);
                        });
                        if (_queue.length === 1) {
                            _executeTop();
                        }
                        return deferred.promise;
                    },
                    /**
                     * After each response completes, unblocks the next request.
                     */
                    response: function (response) {
                        _queue.shift();
                        _executeTop();
                        return response;
                    },
                    /**
                     * After each response errors, unblocks the next request.
                     */
                    responseError: function (responseError) {
                        _queue.shift();
                        _executeTop();
                        return $q.reject(responseError);
                    },
                };
            }]);
        }])
```

Error occurs only in first toaster - then everything is ok.

**Comment by Foxandxss** [comment]
Are you using angular 1.2.x?

**Comment by piernik** [comment]
no angular 1.3.13

**Comment by Foxandxss** [comment]
Could you try the latest .js file on the dist folder on master?

**Comment by piernik** [comment]
The same error :/

**Comment by Foxandxss** [comment]
Uh, what you mean with your app config? and where is the toast on the interceptor? I don't see anything there.

**Comment by piernik** [comment]
ok I have to investigate it myself. But on Monday :)

**Comment by piernik** [comment]
Problem is with one toast (not first as I wrote).
`newToast.scope.init();` is trigered before directive's link function `toastLinkFunction` is executed and that's why I get error.

I don't know why is it happening.

When I remove config I wrote above everything is ok.
Is it possible that $httpProvider is linked to $compile?

**Comment by gauravbhavsar** [comment]
Hi @piernik 
I am also getting the same error, can you please tell me how can you fixed this problem ?

error : newToast.scope.init is not a function.
